src/snek/driver.rs

```diff
@@ -163,3 +171,25 @@ impl From<Key> for UserAction {
     }
   }
 }
+
+#[derive(Debug)]
```

Comment:
can we move the struct definitions to up above the `impl Driver`? Would prefer to have all the type definitions in one place, and all the impls in another